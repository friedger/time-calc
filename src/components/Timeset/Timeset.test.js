import React from 'react'
import ConnectedTimeset, {StyledTimeset} from './Timeset'
import 'jest-enzyme'
import { shallowWithStore, shallowWithState } from 'enzyme-redux'
import { shallow, mount } from 'enzyme'
import { createMockStore } from 'redux-test-utils'
import {Provider} from 'react-redux'

describe('Component - Timeset', () => {
  const date = {start: '12:00', break: '00:30', end: '14:30', duration: '02:00', date: '12/12/2012'}

  it('renders a date set', () => {
    const component = shallow(<StyledTimeset time={date} index={1} onEditTime={jest.fn()} />)

    expect(component.html()).toMatchSnapshot()
  })

  it('triggers the edit callback', () => {
    const cb = jest.fn()
    const component = shallow(<StyledTimeset time={date} index={1} onEditTime={cb} />)

    expect(component.html()).toMatchSnapshot()
    component.find('button[aria-label="Edit"]').simulate('click')

    expect(cb).toHaveBeenCalledWith({})
  })

  it('triggers the delete callback', () => {
    const cb = jest.fn()
    const component = shallow(<StyledTimeset time={date} index={1} onEditTime={jest.fn()} toggleDialog={cb} />)

    expect(component.html()).toMatchSnapshot()
    component.find('button[aria-label="Delete"]').simulate('click')

    expect(cb).toHaveBeenCalledWith({})
  })
})
