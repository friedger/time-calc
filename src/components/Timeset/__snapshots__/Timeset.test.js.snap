// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Timeset renders a date set 1`] = `"<tr class=\\"MuiTableRow-root-4 Component-row-2\\"><td class=\\"MuiTableCell-root-9 Component-important-3\\">December 12, 2012</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">12:00</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">14:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">00:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-important-3\\">02:00</td><td class=\\"MuiTableCell-root-9\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17 MuiIconButton-colorPrimary-19\\" type=\\"button\\" variant=\\"raised\\"><span class=\\"MuiIconButton-label-22\\"><i class=\\"material-icons\\">edit</i></span><span class=\\"MuiTouchRipple-root-26\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17 MuiIconButton-colorSecondary-20\\" type=\\"button\\" variant=\\"raised\\"><span class=\\"MuiIconButton-label-22\\"><i class=\\"material-icons\\">delete</i></span><span class=\\"MuiTouchRipple-root-26\\"></span></button></td></tr>"`;

exports[`Component - Timeset triggers the delete callback 1`] = `"<tr class=\\"MuiTableRow-root-4 Component-row-2\\"><td class=\\"MuiTableCell-root-9 Component-important-3\\">December 12, 2012</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">12:00</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">14:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">00:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-important-3\\">02:00</td><td class=\\"MuiTableCell-root-9\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17 MuiIconButton-colorPrimary-19\\" type=\\"button\\" variant=\\"raised\\"><span class=\\"MuiIconButton-label-22\\"><i class=\\"material-icons\\">edit</i></span><span class=\\"MuiTouchRipple-root-26\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17 MuiIconButton-colorSecondary-20\\" type=\\"button\\" variant=\\"raised\\"><span class=\\"MuiIconButton-label-22\\"><i class=\\"material-icons\\">delete</i></span><span class=\\"MuiTouchRipple-root-26\\"></span></button></td></tr>"`;

exports[`Component - Timeset triggers the edit callback 1`] = `"<tr class=\\"MuiTableRow-root-4 Component-row-2\\"><td class=\\"MuiTableCell-root-9 Component-important-3\\">December 12, 2012</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">12:00</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">14:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-hideMobile-1\\">00:30</td><td class=\\"MuiTableCell-root-9 MuiTableCell-numeric-13 Component-important-3\\">02:00</td><td class=\\"MuiTableCell-root-9\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17 MuiIconButton-colorPrimary-19\\" type=\\"button\\" variant=\\"raised\\"><span class=\\"MuiIconButton-label-22\\"><i class=\\"material-icons\\">edit</i></span><span class=\\"MuiTouchRipple-root-26\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-23 MuiIconButton-root-17 MuiIconButton-colorSecondary-20\\" type=\\"button\\" variant=\\"raised\\"><span class=\\"MuiIconButton-label-22\\"><i class=\\"material-icons\\">delete</i></span><span class=\\"MuiTouchRipple-root-26\\"></span></button></td></tr>"`;
